modules/custom/az_migration/src/Plugin/migrate/process/EntityEmbedProcess.php

```diff
@@ -0,0 +1,297 @@
+    if (!empty($alt)) {
+      $changed->setAttribute('alt', $alt);
+    }
+
+    // Type specific attributes.
+    switch ($type) {
+      case 'file':
+        $changed->setAttribute('data-view-mode', $view);
+        break;
+
+      case 'node':
+        $changed->setAttribute('data-embed-button', "az_embed_content");
+        $changed->setAttribute('data-entity-embed-display', "view_mode:node.{$view}");
+        break;
+    }
+    return $changed;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function transform($value, MigrateExecutableInterface $migrate_executable, Row $row, $destination_property) {
+
+    $dom = new \DOMDocument();
+    $dom->loadHTML($value, LIBXML_HTML_NOIMPLIED | LIBXML_HTML_NODEFDTD);
+    $elements = $dom->getElementsByTagName("drupal-entity");
+
+    // Configuration of custom content.
+    $migrations = self::MIGRATION_MAPPING;
+    $view_modes = self::VIEW_MODES;
+
+    if (!empty($this->configuration['migrations']) && is_array($this->configuration['migrations'])) {
+      $migrations = array_merge($migrations, $this->configuration['migrations']);
+    }
+    if (!empty($this->configuration['view_modes']) && is_array($this->configuration['view_modes'])) {
+      $view_modes = array_merge($view_modes, $this->configuration['view_modes']);
+    }
+
+    // We iterate backwards because it simplifies removal logic.
+    for ($i = $elements->length - 1; $i >= 0; $i--) {
+      $element = $elements->item($i);
+      $type = $element->getAttribute('data-entity-type');
+      $id = $element->getAttribute('data-entity-id');
+      $view = 'default';
+
+      // See if we have a view mode.
+      $settings = $element->getAttribute('data-entity-embed-settings');
+      if (!empty($settings)) {
+        $settings = json_decode($settings, TRUE);
+        $v = $view;
+        // Attempt to map our view mode from QS1 to QS2.
+        if (!empty($settings['view_mode'])) {
+          $v = $settings['view_mode'];
+        }
+        elseif (!empty($settings['file_view_mode'])) {
+          $v = $settings['file_view_mode'];
+        }
+        if (isset($view_modes[$type][$v])) {
+          $view = $view_modes[$type][$v];
+        }
+      }
+
+      $post = NULL;
+      switch ($type) {
+        // Embedded file.
+        case 'file':
+          $post = $this->updateEmbedTag($id, 'media', 'drupal-media', $view, $dom, $element, 'az_media', 'media');
+          break;
+
+        // Embedded node. Special consideration, as we need to know which
+        // migration the node is part of, if any. Migration of an Embedded
+        // node only is defined if it's a type that can be migrated.
+        case 'node':
+          $destination_db = Database::setActiveConnection('migrate');
+          // Lookup of content type.
+          $node_type = Database::getConnection()
+            ->query('SELECT type FROM {node} WHERE nid = :nid', [':nid' => $id])
+            ->fetchField();
+          Database::setActiveConnection($destination_db);
```

Comment:
Nitpick: could we avoid the calls to `Database::setActiveConnection()` by passing an argument to [`Database::getConnection()`](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Database%21Database.php/function/Database%3A%3AgetConnection/9.3.x)?

```suggestion
          // Lookup of content type.
          $node_type = Database::getConnection('migrate')
            ->query('SELECT type FROM {node} WHERE nid = :nid', [':nid' => $id])
            ->fetchField();
```

Reply:
That seems reasonable. I was a bit leery of altering the active connection. A one-time connection that doesn't alter anything seems better. Thanks @joeparsons !